JourneysSquad/JourneysSquad/Model/BillStorage/DoubleLinkedList.swift

```diff
@@ -0,0 +1,61 @@
+/// Represents a linked list data structure.
+struct DoubleLinkedList<T> {
+    /// Inserts a new node with the specified value before the given node.
+    ///
+    /// - Parameters:
+    ///   - value: The value to be inserted.
+    ///   - node: The node before which the new node will be inserted.
+    ///   If `node` is `nil` or equal to the current head node,
+    ///   the new node will be inserted at the beginning of the list.
+    mutating func insert(value: T, before node: DoubleLinkedListNode<T>?) {
+        let newNode = DoubleLinkedListNode(value: value)
+        guard head != nil else {
+            head = newNode
+            tail = newNode
+            return
+        }
+
+        if node === head || node == nil {
+            newNode.next = head
+            head?.previous = newNode
+            head = newNode
+        } else {
+            newNode.next = node
+            newNode.previous = node?.previous
+            node?.previous?.next = newNode
+            node?.previous = newNode
+        }
+    }
+
+    /// Adds a new node with the specified value to the end of the linked list.
+    ///
+    /// - Parameter value: The value to be added to the linked list.
+    mutating func append(value: T) {
+        let newNode = DoubleLinkedListNode(value: value)
+        newNode.previous = tail
+        tail?.next = newNode
+        tail = newNode
+
+        if head == nil {
+            head = newNode
+        }
```

Comment:
Please try to make it simple (please see question about the `tail` above)

Smth like this:

```suggestion
        let newNode = DoubleLinkedListNode(value: value)
        newNode.next = head
        head = newNode
```

?

Hint: you're not required to append new nodes to the end of the list.